### DISCUSSION TITLE
can't use xpm install xpack related toolchain and riscv qemu project template when using a proxy to network.

### DISCUSSION THREAD

**Discussion by shellcreasy** [xpack/xpm-js#192] [Q&A] [answered]
i'm using eclipse embed IDE to create riscv qemu project and the project wizard page show no repsonse when execute "xpm init --template @micro-os-plus/hello-world-qemu-template@1.4.1 --property target=riscv-rv64imafdc --property language=cpp --property buildGenerator=cmake",and can not install the xpack riscv toolchain software neither.The context is that i'm using a company internal proxy to access the network,i do not know how to do so i can successfully download the dependencies that the project needed to create,build and run.

**Comment by ilg-ul** [comment]
Configure proxy for npm, xpm uses the same code to download packages.

**Reply by shellcreasy** [reply]
npm already configured the proxy, still can not download the package,its makes me very upset......

**Reply by ilg-ul** [reply, marked as answer]
Also set `http_proxy` & `https_proxy` in your environment (same content as for `curl`).

If still not working, please run the `xpm init ...` command in a terminal and add `-dd` to get a full trace.

**Reply by shellcreasy** [reply]
thank you so much.i can successfully create,build and debug riscv project after configuring the http and https proxy in .bashrc file.

**Reply by ilg-ul** [reply]
Great! Good luck with your projects!
